browser/ipfs/ipfs_tab_helper_unittest.cc

```diff
@@ -17,12 +17,38 @@
 #include "content/public/test/browser_task_environment.h"
 #include "content/public/test/test_browser_context.h"
 #include "content/public/test/test_renderer_host.h"
+#include "content/public/test/test_web_contents_factory.h"
 #include "content/public/test/web_contents_tester.h"
+#include "content/test/test_web_contents.h"
 #include "net/http/http_response_headers.h"
+#include "services/network/test/test_network_context.h"
 #include "testing/gtest/include/gtest/gtest.h"
 
 namespace ipfs {
 
+class FakeIPFSHostResolver : public ipfs::IPFSHostResolver {
+ public:
+  explicit FakeIPFSHostResolver(network::mojom::NetworkContext* context)
+      : ipfs::IPFSHostResolver(context) {}
+  ~FakeIPFSHostResolver() override = default;
+  void Resolve(const net::HostPortPair& host,
+               const net::NetworkIsolationKey& isolation_key,
+               net::DnsQueryType dns_query_type,
+               HostTextResultsCallback callback) override {
+    resolve_called_++;
+    if (callback)
+      std::move(callback).Run(host.host(), dnslink_);
+  }
+
+  bool resolve_called() const { return resolve_called_ == 1; }
```

Comment:
I think we should just use bool for resolve_called_ then.

Reply:
Fixed